docs/designs/ruby_test_plan_002.md

```diff
@@ -0,0 +1,150 @@
+environment to make it possible to build projects without network connectivity.
+Cachi2 can handle a number of package formats, RubyGems/Bundler (RG/B) is among
+the supported ones. Cachi2 does not support all features of RG/B, opting to
+support the most widespread and safe ones instead of supporting everything.
+
+The tests are to check that conforming packages could be preprocessed by Cachi2
+and that non-conforming packages are rejected by Cachi2.
+
+The checks are expected to be run against actual packages published either on
+rubygems.org, on github.com or locally.
+
+Each example in every test scenario requires a separate repository/package
+demonstrating an aspect under test. A list of necessary repositories could be
+found [below](#necessary-repositories).
+
+## Test cases
+
+@e2e_test
+Feature: Projects prepared with Cachi2 could be hermetically built
+    Scenario: a user can successfully build a project prepared with Cachi2
+        Given a clean repository of a well-defined project
+         When a user request to fetch dependencies for a well-defined project
+          And a user generates environment for well-defined project
+          And a user triggers a hermetic build
+         Then the build system exits successfully
+          And a build artifact is produced.
+
+
+@integration_test
+Feature: Cachi2 can prefetch dependencies for Ruby projects
+    
+    Scenario outline: a user wants to fetch dependencies for a package containing well-defined Gemfile.lock
+       Given  clean repository of a "well-defined project"
+        When  a user request to fetch dependencies for a "well-defined project"
+        Then  direct and transitive dependencies are fetched
+         And  dependencies could be found within the "well-defined project"
+    Examples:
+        |    well-defined project                          |
+        | Depends on gems from rubygems.org only           |
+        | Has correct .gemspec                             |
+        | Has correct Git dependency                       |
+        | Has several correct Git dependencies             |
+        | Has path dependency                              |
+        | Has several path dependencies                    |
+        | Has a mixture of gems, Git and path dependencies |
+
+    Scenario outline: a user wants to fetch dependencies for a package containing ill-defined Gemfile.lock
+       Given  clean repository of a "ill-defined project"
+        When  a user request to fetch dependencies for a "ill-defined project"
+        Then  Cachi2 reports an error
+         And  repository for "ill-defined project" remains unchanged
+    Examples:
+        |    ill-defined project                     |
+        | Gemfile.lock is missing                    |
+        | Gemfile.lock contains critical code        |
+        | Git dependency is not pinned to hash       |
+        | Gem dependency is sourced from unknown URL |
```

Comment:
After some discussion, we decided to provide support for alternative registries: https://github.com/containerbuildsystem/cachi2/blob/a335c10d69885ba1e88e16f767cefce070a222f3/docs/design/rubygems.md?plain=1#L267

Reply:
Thank you, will remove the example.